opentelekomcloud/resource_opentelekomcloud_sfs_file_system_v2.go

```diff
@@ -149,112 +164,112 @@ func resourceSFSFileSystemV2Create(d *schema.ResourceData, meta interface{}) err
-
-OUTER:
-	for key, val := range n.Metadata {
+	metadata := make(map[string]string)
+	for key, val := range share.Metadata {
 		if strings.HasPrefix(key, "#sfs") {
 			continue
 		}
-		for i := range sys_keys {
-			if key == sys_keys[i] {
-				continue OUTER
-			}
+		if strings.Contains(key, "enterprise_project_id") || strings.Contains(key, "share_used") {
+			continue
 		}
-		md[key] = val
+		metadata[key] = val
 	}
-	d.Set("metadata", md)
-
-	rules, err := shares.ListAccessRights(sfsClient, d.Id()).ExtractAccessRights()
+	mErr = multierror.Append(mErr,
+		d.Set("metadata", metadata),
+	)
 
+	rules, err := shares.ListAccessRights(client, d.Id()).ExtractAccessRights()
 	if err != nil {
 		if _, ok := err.(golangsdk.ErrDefault404); ok {
 			d.SetId("")
 			return nil
 		}
 
-		return fmt.Errorf("Error retrieving OpenTelekomCloud Shares: %s", err)
+		return fmt.Errorf("error retrieving OpenTelekomCloud Shares: %s", err)
 	}
 
 	if len(rules) > 0 {
 		rule := rules[0]
-		d.Set("share_access_id", rule.ID)
-		d.Set("access_rule_status", rule.State)
-		d.Set("access_to", rule.AccessTo)
-		d.Set("access_type", rule.AccessType)
-		d.Set("access_level", rule.AccessLevel)
+		mErr = multierror.Append(mErr,
+			d.Set("share_access_id", rule.ID),
+			d.Set("access_rule_status", rule.State),
+			d.Set("access_to", rule.AccessTo),
+			d.Set("access_type", rule.AccessType),
+			d.Set("access_level", rule.AccessLevel),
+		)
 	}
+
+	if mErr.ErrorOrNil() != nil {
+		return mErr
+	}
+
 	return nil
 }
 
 func resourceSFSFileSystemV2Update(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
-	sfsClient, err := config.sfsV2Client(GetRegion(d, config))
+	client, err := config.sfsV2Client(GetRegion(d, config))
 	if err != nil {
-		return fmt.Errorf("Error updating OpenTelekomCloud Share File: %s", err)
+		return fmt.Errorf("error updating OpenTelekomCloud Share File: %s", err)
 	}
 	var updateOpts shares.UpdateOpts
 
 	if d.HasChange("description") || d.HasChange("name") {
 		updateOpts.DisplayName = d.Get("name").(string)
 		updateOpts.DisplayDescription = d.Get("description").(string)
 
-		_, err = shares.Update(sfsClient, d.Id(), updateOpts).Extract()
+		_, err = shares.Update(client, d.Id(), updateOpts).Extract()
 		if err != nil {
-			return fmt.Errorf("Error updating OpenTelekomCloud Share File: %s", err)
+			return fmt.Errorf("error updating OpenTelekomCloud Share File: %s", err)
 		}
 	}
 	if d.HasChange("access_to") || d.HasChange("access_level") || d.HasChange("access_type") {
 		deleteAccessOpts := shares.DeleteAccessOpts{AccessID: d.Get("share_access_id").(string)}
-		deny := shares.DeleteAccess(sfsClient, d.Id(), deleteAccessOpts)
+		deny := shares.DeleteAccess(client, d.Id(), deleteAccessOpts)
 		if deny.Err != nil {
-			return fmt.Errorf("Error changing access rules for share file : %s", deny.Err)
+			return fmt.Errorf("error changing access rules for share file: %s", deny.Err)
```

Comment:
In other places `ExtractErr()` is used, or smth like `if err := shares.DeleteAccess(client, d.Id(), deleteAccessOpts).Err; err != nil...`